corehq/apps/userreports/tasks.py

```diff
@@ -238,9 +239,35 @@ def delete_data_source_task(domain, config_id):
     delete_data_source_shared(domain, config_id)
 
 
-@periodic_task(serializer='pickle',
-    run_every=crontab(minute='*/5'), queue=settings.CELERY_PERIODIC_QUEUE
-)
+@periodic_task(serializer='pickle', run_every=crontab(minute='*/5'), queue=settings.CELERY_PERIODIC_QUEUE)
+def reprocess_archive_stubs():
+    # Check for archive stubs
+    from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+    from couchforms.models import UnfinishedArchiveStub
+    stubs = UnfinishedArchiveStub.objects.filter()
+    datadog_gauge('commcare.unfinished_archive_stubs', len(stubs))
+    start = time.time()
+    cutoff = start + timedelta(minutes=4).total_seconds()
+    for stub in stubs:
+        # Exit this task after 4 minutes so that the same stub isn't ever processed in multiple queues.
+        if time.time() - start > cutoff:
+            return
+        xform = FormAccessors(stub.domain).get_form(form_id=stub.xform_id)
+        # If the history wasn't updated the first time around, run the whole thing again.
+        if not stub.history_updated:
+            if stub.archive:
+                xform.archive(user_id=stub.user_id)
+            else:
+                xform.unarchive(user_id=stub.user_id)
+        # If the history was updated the first time around, just send the update to kafka
+        else:
+            if stub.archive:
+                xform.publish_archive_action_to_kafka(user_id=stub.user_id, archive=True)
+            else:
+                xform.publish_archive_action_to_kafka(user_id=stub.user_id, archive=False)
```

Comment:
nit: just noticed this now but this could be one line:
```
 xform.publish_archive_action_to_kafka(user_id=stub.user_id, archive=stub.archive)
```

Reply:
good point: https://github.com/dimagi/commcare-hq/pull/22470/commits/c629897b1c835844b226f82dd4a3e34c09e6330b